src/cljc/game/core/card.cljc

```diff
@@ -325,7 +325,7 @@
   "Checks if the two cards are the same by :cid. Alternatively specify 1-function to use to check the card"
   ([card1 card2] (same-card? :cid card1 card2))
   ([func card1 card2]
-    (= (func card1) (func card2))))
+   (and (func card1) (func card2) (= (func card1) (func card2)))))
```

Comment:
If we're calling each function twice, let's use a `let` block. Something like:
```suggestion
   (let [r1 (func card1) r2 (func card2)] (and r1 r2 (= r1 r2))))))
```

Reply:
nice catch, thanks!